doc/rst/language/spec/conversions.rst

```diff
@@ -12,79 +12,188 @@ expression can be a type expression. We refer to these two types the
+
+   .. code-block:: chapel
+
+      var a = 1:myInteger;  // cast -- invokes operator :
+
+      var b: myInteger = 2; // initialization -- invokes init=
+
+      var c: myInteger;
+      c = 3;                // split-initialization -- invokes init=
+
+      var d = new myInteger();
+      d = 4;                // assignment -- invokes operator =
+
+   .. BLOCK-test-chapelnoprint
+
+      writeln("a is ", a, " : ", a.type:string);
+      writeln("b is ", b, " : ", b.type:string);
+      writeln("c is ", c, " : ", c.type:string);
+      writeln("d is ", d, " : ", d.type:string);
+
+   .. BLOCK-test-chapeloutput
+
+      a is (intValue = 1) : myInteger
+      b is (intValue = 2) : myInteger
+      c is (intValue = 3) : myInteger
+      d is (intValue = 4) : myInteger
+
+.. _Implicit_Conversion_Call:
+
+Implicit Conversions for Function Calls
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+An implicit conversion for a function call - also called a *coercion* -
+occurs when the actual argument of a function call is converted to the
+type of the corresponding formal argument, if the formal’s intent is
+``param``, ``in``, ``const in``, or an abstract intent
+(:ref:`Abstract_Intents`) with the semantics of ``in`` or ``const in``.
+
+Additionally, the formal argument is considered to apply - without making
+a copy of the actual argument - when the actual type is a subtype of the
+formal type. This rule applies to ``in``, ``const in``, ``const ref``,
+and ``type`` intent formals and includes generic formal types.
+See:ref:`Subtype_Arg_Conversions`.
+
+Implicit conversions are not applied for actual arguments passed to
+``ref`` formal arguments.
+
+   *Open issue*.
+
+   Subtype conversions are not yet implemented for formals with
+   ``const ref`` intent. However this feature is planned.
+
+   *Open issue*.
+
+   Should Chapel allow user-defined implicit conversions for function
+   calls?  If so, how would the user define them?
+
+.. _Implicit_Conversion_Conditionals:
+.. _Implicit_Statement_Bool_Conversions:
+
+Implicit Conversions for Conditionals
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+An implicit conversion for a conditional occurs at each of the following
+program locations:
 
 -  The condition of a conditional expression, conditional statement,
    while-do or do-while loop statement is converted to the boolean type.
-   See :ref:`Implicit_Statement_Bool_Conversions`.
 
-Implicit conversions are not applied for actual arguments passed to
-``ref`` or ``const ref`` formal arguments.
+In such a condition, the following implicit conversions to ``bool`` are
+supported:
 
-Implicit conversions *are allowed* between the following source and
-target types, as defined in the referenced subsections:
+-  An expression of integral type is taken to be ``false`` if it is ``0`` and
+   is ``true`` otherwise.
+
+-  An expression of a class type is taken to be ``false`` if it is ``nil`` and
+   is ``true`` otherwise.
```

Comment:
Given that we define _cond_test for reMatch, c_ptr, byteIndex, syserr, we can leave this list open, ex. "- Certain other standard types define such a conversion" and something like "as indicated in their respective sections".